moVirt/src/main/java/org/ovirt/mobile/movirt/rest/VvFileHttpMessageConverter.java

```diff
@@ -0,0 +1,121 @@
+package org.ovirt.mobile.movirt.rest;
+
+/**
+ * Created by suomiy on 10/17/16.
+ */
+
+import android.util.Log;
+
+import org.ovirt.mobile.movirt.model.ConsoleProtocol;
+import org.ovirt.mobile.movirt.rest.dto.ConsoleConnectionDetails;
+import org.springframework.http.HttpInputMessage;
+import org.springframework.http.HttpOutputMessage;
+import org.springframework.http.MediaType;
+import org.springframework.http.converter.AbstractHttpMessageConverter;
+import org.springframework.http.converter.HttpMessageNotReadableException;
+import org.springframework.http.converter.HttpMessageNotWritableException;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.charset.Charset;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Scanner;
+import java.util.regex.MatchResult;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class VvFileHttpMessageConverter extends AbstractHttpMessageConverter<Object> {
+    private static final String TAG = VvFileHttpMessageConverter.class.getSimpleName();
+
+    public static final Charset DEFAULT_CHARSET = Charset.forName("UTF-8");
+    public static final String X_VIRT_VIEWER_MEDIA_TYPE = "application/x-virt-viewer";
+
+    public VvFileHttpMessageConverter() {
+        super(new MediaType("application", "x-virt-viewer", DEFAULT_CHARSET));
+    }
+
+    @Override
+    protected Object readInternal(Class<? extends Object> clazz,
+                                  HttpInputMessage inputMessage) throws IOException, HttpMessageNotReadableException {
+        Object result = null;
+        InputStream inputStream = inputMessage.getBody();
+        try {
+            result = convertStreamToConsoleConnectionDetails(inputStream);
+        } finally {
+            if (inputStream != null) {
```

Comment:
can the inputStream be null? If yes, than what will happen to convertStreamToConsoleConnectionDetails if you pass it null?